### PR TITLE
[Mapping] Start map_playground endpoint and MapPlayground components

### PR DESCRIPTION
**Pull request by jshoe**
### Description
- Starts a locations/map_playground endpoint that will return currently viewable values for MetadataField collection_location. Eventually there will be new Location objects to use instead.
- Endpoint tests in locations_controller_test.rb.
- Starts MapPlayground and BaseMap React components.
- Includes setup for react-map-gl and tiles from MapTiler.
- NPM changes were produced by running `npm install react-map-gl`

### Demo
- Will look like this (but more zoomed in)
Screen Shot 2019-04-15 at 5 58 07 PM

### Test and Reproduce
- Pull the branch, retrieve MAPTILER_API_KEY from Parameter Store, start locally with `MAPTILER_API_KEY=key_here docker-compose up web`
- Visit http://localhost:3000/locations/map_playground
- See map displayed and explore

### CODE REVIEW

File: test/controllers/locations_controller_test.rb [hunk 1]
```diff
@@ -67,4 +67,23 @@ class LocationsControllerTest < ActionDispatch::IntegrationTest
     assert_response :error
     assert_equal JSON.dump(@api_key_error), @response.body
   end
+
```
Comment (cdebourcy) [inline test/controllers/locations_controller_test.rb]: Could you write some "joe CANNOT..." type access control tests for the new `map_playground` endpoint?
"joe cannot see collection location for another user's private sample in map playground"

File: app/assets/src/components/views/discovery/BaseMap.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        zoom
+      }
+    };
+  }
+
+  render() {
+    const { mapTilerKey, renderMarker, markers } = this.props;
+    const { viewport } = this.state;
+
+    const styleID = "a63d8846-0896-4fd0-b062-c9d992a49b63";
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/discovery/BaseMap.jsx]: Is this key not sensitive?

File: app/controllers/locations_controller.rb [hunk 3]
```diff
@@ -31,6 +31,24 @@ def external_search
     }, status: :internal_server_error
   end
 
+  def map_playground
+    # Show all viewable locations
+    field_id = MetadataField.find_by(name: "collection_location")
+    viewable = current_power.samples.pluck(:id)
```
Comment (tfrcarvalho) [inline app/controllers/locations_controller.rb]: As discussed, we should try to stop using large lists of sample ids, in lieu of using the actual conditions to the database. In this case, something like:
```
samples_with_metadata = samples
                            .includes(metadata: :metadata_field)
                            .where(metadata: { metadata_field_id: metadata_field.metadata_field_id })
```

File: app/assets/src/components/views/discovery/BaseMap.jsx (hunk 2 above)
Comment (tfrcarvalho) [reply app/assets/src/components/views/discovery/BaseMap.jsx]: Never mind... seems like it is just a public id.

File: test/controllers/locations_controller_test.rb (hunk 1 above)
Comment (jshoe) [reply test/controllers/locations_controller_test.rb]: 👍

File: app/controllers/locations_controller.rb (hunk 3 above)
Comment (jshoe) [reply app/controllers/locations_controller.rb]: 👍

**Pull request merged**